boa_engine/src/builtins/function/mod.rs

```diff
@@ -498,16 +499,15 @@ impl BuiltInFunctionObject {
             } else {
                 let mut parameters = Vec::with_capacity(args.len());
                 for arg in args {
-                    parameters.push(arg.to_string(context)?);
+                    parameters.push(arg.to_string(context)?.as_slice().to_owned());
                 }
-                let mut parameters = parameters.join(",");
-                parameters.push(')');
-
-                let parameters = match Parser::new(parameters.as_bytes()).parse_formal_parameters(
-                    context.interner_mut(),
-                    generator,
-                    r#async,
-                ) {
+                let mut parameters = parameters.join(utf16!(","));
+                parameters.push(u16::from(b')'));
+
+                // TODO: make parser generic to u32 iterators
```

Comment:
Shouldn't it be generic to `u16` iterators?

Reply:
Not really, since we would need to convert any source code to UTF-16 before passing it to the parser. On the other hand, we can convert both UTF-16 and UTF-8 strings to u32 codepoint iterators without allocating additional space.